### PR TITLE
feat: add algorithm 95 to find the longest amicable chain

### PR DESCRIPTION
**Pull request by abda-gaye**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by abda-gaye** [comment]
This PR adds a solution for Project Euler Problem 95, which finds the smallest member of the longest amicable chain under one million.

Changes:

Implemented find_longest_amicable_chain to calculate amicable chains using proper divisors.
Added docstrings for clarity.
Successfully validated the solution against the problem requirements.

File: project_euler/problem_095/sol1.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    # Loop through possible divisors from 2 to the square root of n
+    for i in range(2, sqrt_n + 1):
+        if n % i == 0:  # Check if i is a divisor of n
+            total += i  # Add the divisor
+            if i != n // i:  # Avoid adding the square root twice
+                total += n // i  # Add the corresponding divisor (n/i)
+    
+    return total
+
+def find_longest_amicable_chain(limit):
```
[the remark this answers is not in the document]
Comment (abda-gaye) [reply project_euler/problem_095/sol1.py]: it's okay

File: project_euler/problem_095/sol1.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            cycle_start_index = chain.index(current)  # Find where the cycle starts
+            if current in sum_divisors and sum_divisors[current] in chain:
+                # This means we have a valid amicable chain
+                chain = chain[cycle_start_index:]  # Take only the cycle part
+                if len(chain) > len(longest_chain):
+                    longest_chain = chain  # Update longest chain if this one is longer
+
+    return min(longest_chain) if longest_chain else None  # Return the smallest member of the longest chain
+
+def solution():
```
[the remark this answers is not in the document]
Comment (abda-gaye) [reply project_euler/problem_095/sol1.py]: it's okay

**Comment by cclauss** [comment]
Closing require_type_hints PRs to prepare for Hacktoberfest

**Pull request closed without merging**